src/PriorityReliableQueue.php

```diff
@@ -0,0 +1,92 @@
+<?php
+namespace Altmetric;
+
+use Redis;
+use Psr\Log\LoggerInterface;
+
+class PriorityReliableQueue implements \Iterator
+{
+    public $name;
+    public $queues;
+    private $redis;
+    private $logger;
+    private $value;
+    private $queue;
+    private $workingQueue;
+
+    public function __construct($name, array $queues, Redis $redis, LoggerInterface $logger)
+    {
+        $this->name = $name;
+        $this->redis = $redis;
+        $this->logger = $logger;
+
+        foreach ($queues as $queue) {
+            $this->queues[$queue] = "{$queue}.working_on.{$name}";
+        }
+    }
+
+    public function rewind()
```

Comment:
Yes, take a look at the [`Iterator` interface](http://php.net/manual/en/class.iterator.php), it's a key part of how `foreach` works:

In short, when you loop over an iterator, it calls the following methods in order:

1. `rewind`
2. `valid`
3. `current`
4. `key`
5. `next`
6. Go to 2.